readthedocs/core/resolver.py

```diff
@@ -147,23 +147,25 @@ def resolve(self, project, protocol='http', filename='', private=None,
         canonical_project = self._get_canonical_project(project)
         custom_domain = self._get_project_custom_domain(canonical_project)
 
-        # This duplication from resolve_domain is for performance purposes
-        # in order to check whether a custom domain should be HTTPS
-        if custom_domain:
+        if self._use_custom_domain(custom_domain):
             domain = custom_domain.domain
         elif self._use_subdomain():
             domain = self._get_project_subdomain(canonical_project)
         else:
             domain = getattr(settings, 'PRODUCTION_DOMAIN')
 
-        if custom_domain:
-            protocol = 'https' if custom_domain.https else 'http'
-        else:
-            # Use HTTPS if settings specify
-            public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
-            use_https = getattr(settings, 'PUBLIC_DOMAIN_USES_HTTPS', False)
-            if use_https and public_domain and public_domain in domain:
-                protocol = 'https'
+        public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
+        use_https = getattr(settings, 'PUBLIC_DOMAIN_USES_HTTPS', False)
+
+        use_https_protocol = any([
+            # Rely on the ``Domain.https`` field
+            custom_domain and custom_domain.https,
```

Comment:
I believe the logic on this line should be `self._use_custom_domain(custom_domain) and custom_domain.https`. Correct?

Reply:
I mean, functionally, it works but from a logic perspective I think the above is better.